src/ol/style/expressions.js

```diff
@@ -784,3 +785,58 @@ Operators['case'] = {
     return result;
   },
 };
+Operators['in'] = {
+  getReturnType: function (args) {
+    return ValueTypes.BOOLEAN;
+  },
+  toGlsl: function (context, args) {
+    assertArgsCount(args, 2);
+
+    const haystack = [];
+    const needle = [];
+    let result;
+    if (getValueType(args[1]) & ValueTypes.NUMBER_ARRAY) {
+      const keyword = expressionToGlsl(context, args[0]);
+      const input = args[1];
+      assertNumbers(input);
+      let i = 0;
+      let inputArr = ['array'];
+      // @ts-ignore
+      input.forEach((element) => {
+        inputArr.push(element);
+        i++;
+        if (i % 4 === 0) {
+          haystack.push(expressionToGlsl(context, inputArr));
+          needle.push(`vec4(${keyword})`);
+          inputArr = ['array'];
+        }
+      });
+      if (i % 4 === 1) {
+        inputArr.push(inputArr[1]);
+        haystack.push(expressionToGlsl(context, inputArr));
+        needle.push(`vec${inputArr.length - 1}(${keyword})`);
+      } else {
+        haystack.push(expressionToGlsl(context, inputArr));
+        needle.push(`vec${inputArr.length - 1}(${keyword})`);
+      }
+      const preliminary = [];
+      for (i = 0; i < haystack.length; i++) {
+        preliminary.push([`any(equal(${haystack[i]}, ${needle[i]}))`]);
+      }
+      result = preliminary.join(` || `);
+    } else if (getValueType(args[1]) & ValueTypes.STRING) {
+      const keyword = args[0];
+      const input = args[1];
+      assertString(input);
+      assertString(keyword);
+      // @ts-ignore
```

Comment:
```suggestion
      assertString(args[0]);
      const keyword = /** @type {string} */ (args[0]);
      const input = /** @type {string} */ (args[1]);
```
Same as above, assertString is not necessary for input because it has already been checked and casting the variables gets rid of the ts warnings.